tests/sync/app.cpp

```diff
@@ -1811,4 +1807,25 @@ TEST_CASE("app: auth providers", "[sync][app]") {
         CHECK(credentials.provider_as_string() == IdentityProviderUsernamePassword);
         CHECK(credentials.serialize_as_json() == "{\"password\":\"pass\",\"provider\":\"local-userpass\",\"username\":\"user\"}");
     }
+    
+    SECTION("auth providers function") {
```

Comment:
Missing a test for a function with arguments.

Reply:
Also, I'm a bit worried that there is no test for this against the server, as it seems a fairly complex piece of functionality.

Reply:
We test this in the current SDKs, I agree with @cmelchior , we should absolutely be testing this.

Reply:
Test has been added